Migrant Waves Review: Bigger, Better, Bupdated

Ruediger4 opened the pull request:
## About The Pull Request
**MIGRANT WAVES**! Boy how do we love those when they're thematic.

Sadly a LOT of them fell out of favor/touch. I'm talking no sword/dagger sheathes, weird design choices, stats not making sense.

What originally started as "I need to review the Zyb Slavers" became "I'll check out Heartfelt too" and into "fuck it, I'll readd the Gronnian guys".

Needless to say, the Gronnian part/credits was made back in RW1 and the credits go to ProbablyCarl, I merely ported them! https://github.com/Rotwood-Vale/Ratwood-Keep/pull/2225

## WHAT THE PR DOES:
- **Heartfelt wave** modifications. Add one wave that allows the Prior to appear with the Knight and the Lord.
- **Heartfelt roles** modifications. Some stats, some skills, give the Knight RNG when it comes to getting his weapon (longsword still included), force the Prior to be Undivided, give the Magos a diamond staff.
- **Zybantine wave** modifications! They get a wave where its 1 of each goon (1 master, blade guy, crossbow guy, whip guy) and **no slave**. The idea? Go get one!
- **Zybantine roles** modifications. Gives the Master a proper Zyb Sallet (Jannisaries use them), modify some of their equipment, make Whip Slaver better by giving him a shield, and a bit more that I forgot by now.
- **Fablefield roles and waves** modifications! Now you can be a solo Gouliard! Your goons get a bit more sword skill! They also get random instruments!
- **Readds the Gronn wave** with some modifications in stats, skills, spells for the Shaman (no more magics, now you get proper miracles), SOME rng for the warriors because I like RNG. **Most of the armor is in for the Steppesmen**, so they'll be using that.
<!-- Describe your pull request. Avoid text walls, use concise bullet points for easier readability. Document every change, or this can delay review and even discourage maintainers from merging your PR. -->

## Testing Evidence
**GRONN**
One of the Warrior Drips
<img width="218" height="169" alt="image" src="https://github.com/user-attachments/assets/f48745bf-bba3-4dc1-b0ce-c853a7ae9fb6" />
Shaman and the Graggah Spells
<img width="571" height="592" alt="image" src="https://github.com/user-attachments/assets/958b37af-bc1c-4493-a825-4983b8a150e9" />
The wave working as intended.
<img width="322" height="397" alt="image" src="https://github.com/user-attachments/assets/8730f700-ce11-495b-a58a-3706c7027e5f" />

**HEARTFELT**
Knight's Gambling In Action
<img width="268" height="516" alt="image" src="https://github.com/user-attachments/assets/866d369c-1239-4ba2-9079-6c2c42c76200" />
<img width="314" height="491" alt="image" src="https://github.com/user-attachments/assets/06066590-8bf6-433d-997f-574538ed1360" />
Magos & their staff
<img width="317" height="569" alt="image" src="https://github.com/user-attachments/assets/1f014042-f487-42e9-8e9f-6da29092cf4c" />

**ZYBANTINE SLAVERS**
Whip Merc & Their Gear
<img width="346" height="585" alt="image" src="https://github.com/user-attachments/assets/f7fa858e-2825-4cc1-8230-aa0fba94a0ce" />
<img width="121" height="119" alt="image" src="https://github.com/user-attachments/assets/b69ed755-4052-4f75-9935-b1cdaf36e9ec" />
Master Slaver
<img width="297" height="193" alt="image" src="https://github.com/user-attachments/assets/bdaf1ed8-7372-42f3-90f4-6cac2a6e60a5" />
<img width="313" height="453" alt="image" src="https://github.com/user-attachments/assets/b741d903-27ec-4280-a8ec-a082d3fb5d29" />

**FABLEFIELD BARDS**
RNG Bard Instrument
<img width="333" height="576" alt="image" src="https://github.com/user-attachments/assets/7f04511c-ddcb-4ca9-801e-5dc8ceb6814b" />
<img width="310" height="548" alt="image" src="https://github.com/user-attachments/assets/e9d3dda2-6c86-4662-a46e-ea2786214fd1" />


<!-- It's mandatory to test your PR. Provide images, clips or description of how you tested your changes where possible. -->

## Why It's Good For The Game
Migrant waves can actually be fun to play when they're not Extremely Dogshit or outdated, methinks!
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial. If you can't, then it probably isn't good for the game in the first place. -->

<!-- By contributing to this codebase, you confirm that any code and sprites you provide are legal to share and will be licensed under the terms specified in README.md — AGPLv3 for code and CC-BY-SA 3.0 for assets, unless otherwise stated. You acknowledge that the project maintainers are under no obligation to remove any materials that do not violate these licenses. -->

snowysquirrel commented:
> NOOOOO NOT THE AVAR-FUNDED GRONNITES

gronn has fallen on hard times ...

Ruediger4 commented:
> also noticed that shaman is given prestidigitation instead of orison as a starter? or is orison given automatically since they have holy skill?

They get Orison as shown in their screenshot. Left predigestation in for silly "Holy Shit He Can Snap His Fingers To Make A Fire!" thing + not relying on lampterns

> perhaps change the gronn migration roles to start with gronn language, instead of orcish?
> 
> the current atgervi/gronnic mercenaries are locked to inhumen pantheons in general instead of just graggar, i'd personally enjoy the option of choosing which i am for the gronn waves
> 
> we should look at the allowed races for the gronn waves as well -- i do think some sense of homogeneity is good since this is a tribe/warband, but a few flavorful additions could be good too adding half-kin, maybe half-elf?, makes sense since gronnic tribes take slaves from raids and the like

I actually Forgor about the languages thing. Good point! I'll make the change.

As for the Inhumen Pantheon thing - I'd say it depends on where the spirit of the wave relies on. These are raiders. These people come from Gronn to pillage, loot, probably burn down an orphanage or two. I *think* it'd be fine to keep them Graggar-only as it's, again, thematic!

And aye, we were talking about the species in dev-general. May be looking at expanding it for some/most? furry races (i.e. Lupian, Tabaxi, Sisseans/Zardmen) and maybe even dwarves and half/wildkins. **The reason I am not moving my hand too much is because the Gronn wave was originally Carl and Chocobo's thing**, and I sense it'd be kinda mean to just go "Oh teehee they're furry raiders now :]"

> NOOOOO NOT THE AVAR-FUNDED GRONNITES

That's on whoever took the sprites in AP and turned them into Steppesmen drip! I'll prolly modify the descriptions to fit the fact they're also used by Gronnians as well.

Hmeister-real commented on code/datums/migrants/migrant_waves/slaver roles.dm:
```diff
@@ -7,13 +7,13 @@
 
 /datum/outfit/job/roguetown/slaver/master/pre_equip(mob/living/carbon/human/H)
```
should give Master shield skill and/or that one shield that takes the skill of your weapon considering he uses whips

Hmeister-real replied:
Thanks for updating slavers for 2.0 btw, was planning to do that myself at some point

Ruediger4 replied:
> should give Master shield skill and/or that one shield that takes the skill of your weapon considering he uses whips

You've no idea how jacked up shield skills are.

Slaver gets *master* whips, a better whip, and a really beefy statblock. I thought about it but I think it's better if they don't. 

They're slavers, not warlords. If he is Solo Bolo, he needs to be smart about it (either get a weaker prey or hire more goons or get a sword/buckler on his own)

Whip merc gets it because they don't have a jacked up statblock or master whip skill. And because you need a 'larger' party for them to show up and be reliable.

Ruediger4 commented:
Recent changes:
* Troubleshooted and fixed?? the issue where you couldn't examine the migrant wave roles. Praying it works dandy and all. It works on my PC. Requested by @ProbablyCarl 
<img width="453" height="173" alt="image" src="https://github.com/user-attachments/assets/29aab657-c19a-41fe-a2ee-9883639b5e7f" />
<img width="461" height="471" alt="image" src="https://github.com/user-attachments/assets/2ea77984-6b3f-4ccd-8363-966eb909ce86" />

* **ZYB ROLES** get only 1 point in medicine. Crossbow fellow still gets 3, slave gets 2. You can stitch some wounds but the crossbow guy is the dedicated healslut.
* **ZYB BLADE AND WHIP MERC** now start with 1 less coin pouch (idk why they had 2) in favor of a steel chain coif. Rejoice.
* **ZYB CROSSBOW DUDE** gets 5 skill crossbow since most (if not all) dedicated crossbow classes get that. They lose weapon variety a bit (no axe in case someone wants to grind up axe skill w/the LITERAL MASTER SKILL SLAVEMASTER). **Crossbow skill affects reload and aim time, and this goon gets no STR points to reload any faster**.
* **ZYB MASTER** gets expert wrestling so he can pin new slaves faster. Still unarmed 3 so dont go karate kid'ing.
* **HEARTFELT KNIGHT** has a 1 in 5 to get an Eagle's Beak instead of a Lucerne.
* **GRONN WAVE 2** gets 2 warriors instead of 1 (so the spread is 1 chieftain, 1 shaman, 2 warriors, 2 tribals, 1 slave).
* **GRONN CHIEFTAIN** gets an iron gorget. **GRONN SHAMAN** gets a leather gorget and iron bracers to unarmed parry with.
* Removed Herobrine
* No I'm not giving shield skills to the Slavemaster. He gets +14 stat points total **without virtues or statpacks**, on top of master whips. Go buy a buckler or parry w/a sword in your offhand.

snowysquirrel commented:
has anyone noticed a gronn migrant wave pop up as an option? i was checking a lot today out of curiosity, and got a lot of zybantine offers but no gronn
